### ISSUE TITLE
Quitting magit-status is slow

### ISSUE THREAD

**Issue by ijacquez** [hlissner/doom-emacs#1298]
### Observed behavior

`magit-status` closes as expected, but there is about a 5-15 second delay (Mac OS spinning pinwheel). My suspicion is that `(define-key magit-status-mode-map [remap magit-mode-bury-buffer] #'+magit/quit)` (`+magit/quit`) in `.emacs.d/modules/tools/magit/config.el` has something to do with it.

### Expected behavior

No delay after quitting `magit-status`.

### Steps to reproduce

1. Open 150+ buffers (C# files?)
2. `C-c v g` to open `magit-status`
3. Quit `magit-status` with `q`

### System information
1. GNU Emacs 26.1 (build 1, x86_64-apple-darwin18.0.0, NS appkit-1671.00 Version 10.14 (Build 18A391)) of 2018-10-04
2. Doom hash: 99aefd6e022fc092c3c8e22a3c42b78cec0823d2
3. `evil` feature is disabled.

**Comment by UndeadKernel** [comment]
With 150+ buffers open, magit will slow down when closing the status buffer, indeed. This is because Doom will go over every open buffer to update these.

This is done so that the gutters (and modelines) of every buffer are in sync with whatever you have done in Magit. Doom also kills all other left over Magit windows. If you do not care about syncing the gutters, the modeline and potentially leaving unwanted magit buffers in the background, you can disable Dooms cleanup/update process with:
```lisp
(define-key magit-status-mode-map [remap magit-mode-bury-buffer] nil)
```

**Comment by hlissner** [comment]
This should be fixed as of bdd410c. I've employed a new strategy with `+magit/quit` which should be faster; it marks buffers to be refreshed when you next switch to them. Please update and let me know if it's more tolerable now.

**Comment by ijacquez** [comment]
I just updated. And so far, it's much, much faster. Thank you. I'll create a new issue (or re-open this one) if a new related issue arises. Thanks.
